contracts/SimpleMarginTrading.sol

```diff
@@ -116,66 +116,69 @@ contract SimpleMarginTrading
     function open(ZeroExQuote memory quote)
         public
         payable
-        onlyOwner
+        onlyowner
         onlyWhenClosed
         returns (uint256 positionBalance, uint256 borrowBalance)
     {
         // 1. increase position by msg.value - protocolFee
         positionBalance = msg.value.safeSub(quote.protocolFee);
         // 2. mint collateral in compound
-        CETH.mint.value(positionBalance)();
+        ceth.mint.value(positionBalance)();
         // 3. borrow token
-        require(CDAI.borrow(quote.sellAmount) == 0, "borrow didn't work");
-        // 4. swap token for collateral
-        _approve(address(DAI), _getZeroExApprovalAddress());
-        // 5. execute swap
-        (bool success, bytes memory data) = address(EXCHANGE).call.value(quote.protocolFee)(quote.calldataHex);
+        require(Cdai.borrow(quote.sellAmount) == 0, "borrow didn't work");
+        // 4. approve 0x exchange to move DAI
+        _approve(address(dai), _getZeroExApprovalAddress());
+        // 5. verify quote is valid
+        require(quote.sellToken == address(weth), "not selling weth");
+        require(quote.buyToken == address(dai), "not buying dai");
+        // 6. execute swap
+        (bool success, bytes memory data) = address(exchange).call.value(quote.protocolFee)(quote.calldataHex);
         require(success, "Swap not filled.");
-        // 6. decode fill results
+        // 7. decode fill results
         LibFillResults.FillResults memory fillResults = abi.decode(data, (LibFillResults.FillResults));
-        // 7. position size increase by bought amount of WETH
-        positionBalance += fillResults.makerAssetFilledAmount;
-        borrowBalance = CDAI.borrowBalanceCurrent(address(this));
-        // at this point you have CETH, and swapped for WETH
+        // 8. position size increase by bought amount of WETH
+        positionBalance.safeAdd(fillResults.makerAssetFilledAmount);
+        borrowBalance = Cdai.borrowBalanceCurrent(address(this));
+        // at this point you have ceth, and swapped for WETH
     }
 
     function close(
        ZeroExQuote memory quote
     )
         public
-        onlyOwner
+        onlyowner
         onlyWhenOpen
         returns (uint ethBalance)
     {
         // 1. approve for swap
-        _approve(address(WETH), _getZeroExApprovalAddress());
+        _approve(address(weth), _getZeroExApprovalAddress());
         // 2. verify swap
-        uint256 wethBalance = WETH.balanceOf(address(this));
-        uint256 daiBorrowBalance = CDAI.borrowBalanceCurrent(address(this)); // TODO
+        uint256 wethBalance = weth.balanceOf(address(this));
+        uint256 daiBorrowBalance = Cdai.borrowBalanceCurrent(address(this)); // TODO
         require(wethBalance > quote.sellAmount, "not enough to swap");
```

Comment:
This should probably be `>=` right? A `wethBalance` of 1 will get stuck in this contract.